vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/samples/batchRegistration/SampleRegistrationDialog.java

```diff
@@ -0,0 +1,166 @@
+package life.qbic.datamanager.views.projects.project.samples.batchRegistration;
+
+import com.vaadin.flow.component.ComponentEventListener;
+import com.vaadin.flow.component.avatar.Avatar;
+import com.vaadin.flow.component.dialog.Dialog;
+import com.vaadin.flow.component.html.Span;
+import com.vaadin.flow.component.tabs.Tab;
+import com.vaadin.flow.component.tabs.TabSheet;
+import com.vaadin.flow.component.tabs.TabVariant;
+import com.vaadin.flow.theme.lumo.LumoUtility.Margin.Left;
+import com.vaadin.flow.theme.lumo.LumoUtility.Margin.Top;
+import java.util.ArrayList;
+import java.util.List;
+import life.qbic.datamanager.views.events.UserCancelEvent;
+import life.qbic.projectmanagement.application.SampleInformationService.Sample;
+import life.qbic.projectmanagement.application.SampleRegistrationService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+/**
+ * <b>Sample Registration Dialog</b>
+ *
+ * <p>Component to register {@link Sample} with their associated metadata information</p>
+ */
+public class SampleRegistrationDialog extends Dialog {
+
+  private final Span title = new Span("Upload Laboratory Metadata");
+  private final TabSheet tabStepper = new TabSheet();
+  private final Tab generalInformationTab = createTabStep("1", "General Information");
+  private final Tab sampleMetadataTab = createTabStep("2", "Sample Metadata");
+  private final GeneralInformationLayout generalInformationLayout = new GeneralInformationLayout();
+  private SampleSpreadsheetLayout sampleSpreadsheetLayout;
+  private final RegisterBatchDialogHandler registerBatchDialogHandler;
+
+  public SampleRegistrationDialog(@Autowired SampleRegistrationService sampleRegistrationService) {
+    add(title);
+    title.addClassNames("text-2xl", "font-bold", "text-secondary");
+    initSampleMetadataLayout(sampleRegistrationService);
+    initTabStepper();
+    styleStepper();
+    registerBatchDialogHandler = new RegisterBatchDialogHandler();
+    this.setSizeFull();
+  }
+
+  private void initTabStepper() {
+    tabStepper.add(generalInformationTab, generalInformationLayout);
+    tabStepper.add(sampleMetadataTab, sampleSpreadsheetLayout);
+    add(tabStepper);
+  }
+
+  private Tab createTabStep(String avatarLabel, String tabLabel) {
+    Avatar stepAvatar = new Avatar(avatarLabel);
+    stepAvatar.setColorIndex(2);
```

Comment:
what does this do? seems very hardcoded

Reply:
Sets the stepper color which is currently staying the same independent of selection